source/appModules/eclipse.py

```diff
@@ -17,14 +22,83 @@ def event_valueChange(self):
 		# Therefore, just drop this event.
 		pass
 
+class AutocompletionListView(IAccessible):
+
+	def _get_name(self):
+		return AUTOCOMPLETION_LIST_NAME
+
+class AutocompletionListItem(IAccessible):
+
+	def event_selection(self):
+		# Fixme: I don't know if it is the correct way to do this.
+
+		# I think that you can feel more confortable if the text editor gain the focus after the item becomes hidden.
+		if self.appModule.textEditor == None:
+			self.appModule.textEditor = api.getFocusObject()
+
+		if self.appModule.selectedItem != self:
+			self.appModule.selectedItem = self
+			api.setFocusObject(self)
+			self.reportFocus()
+
+	def script_returnToTextEditor(self, gesture):
+		# We need to send the gesture to hide the autocompletion window
+		gesture.send()
+
+		# Retrieve the focus to the editor
+		if self.appModule.textEditor:
+			api.setFocusObject(self.appModule.textEditor)
+			self.appModule.textEditor.reportFocus()
+			
+			# And prevent confusing events
+			self.appModule.textEditor = None
+		
+		# And clean up the selected list item because we don't need it
+		self.appModule.selectedItem = None
+
+	def script_readDocumentation(self, gesture):
+		obj = api.getDesktopObject().simpleFirstChild

```

Comment:
Hi,

El 17/01/2016 a las 11:49 p.m., James Teh escribió:
> In source/appModules/eclipse.py
> <https://github.com/nvaccess/nvda/pull/5680#discussion_r49958557>:
>
>> +		# We need to send the gesture to hide the autocompletion window
>> +		gesture.send()
>> +
>> +		# Retrieve the focus to the editor
>> +		if self.appModule.textEditor:
>> +			api.setFocusObject(self.appModule.textEditor)
>> +			self.appModule.textEditor.reportFocus()
>> +			
>> +			# And prevent confusing events
>> +			self.appModule.textEditor = None
>> +		
>> +		# And clean up the selected list item because we don't need it
>> +		self.appModule.selectedItem = None
>> +
>> +	def script_readDocumentation(self, gesture):
>> +		obj = api.getDesktopObject().simpleFirstChild
>
> You should generally avoid using simple* properties in plugins. They can
> be expensive, can cause infinite recursion in some cases and might
> change in future NVDA versions. The Developer Guide does warn against
> this, though the warning could probably be stronger. Instead, figure out
> the correct path using the raw .firstChild, .next, etc. properties.

OK, Thanks for the suggestion. For now, it's a basic implementation of 
the script. It works, but needs some fixes and additions.

I'll see if I can change this, because sometimes it slows down NVDA's 
performance.

> I'm also concerned about going from the desktop object like this. There
> could be cases where the first object is not even associated with your
> application. I assume this is some sort of floating window?

Yep, it's a floating window, and the Developer Guide have only basic 
documentation and I didn't read the sources. I'll try to fix it.

Thanks to all for your help.

Have a nice day.
-- 
¡Cuando tus fuerzas terminan, las de Dios comienzan!